### ISSUE TITLE
Example for fMP4

### ISSUE THREAD

**Issue by miluxmilux** [alfg/mp4-rust#36]
Hi there!
It seems, by looking at the code, that fragmented mp4 boxes are supported. Do you happen to have an example by chance? Or would this need something more to be implemented?
Thanks!

**Comment by alfg** [comment]
Hi @miluxmilux, 

The fragmented boxes: mehd, moof, mvex, tfhd, traf, trex, trun should be supported for reading. However, writing is not fully implemented yet.

**Comment by hansihe** [comment]
@miluxmilux Interestingly I have also been working on this for the last week.

I have a fork that is able to generate segments that plays fine in both Firefox and Chrome through MSE.

I'll push up my code in a bit if you're interested in having a look at my code.

-------

@alfg I'll push up my changes once I sort out a few remaining issues. I had to make some fairly major changes in some places and add a few boxes that's needed for browser and MPEG-DASH spec compatibility.

I have also exposed the raw box api externally since I don't have the time or familiarity with MP4 to make a general abstraction for everything.

Would be cool if we could upstream at least part of my changes.

**Comment by alfg** [comment]
@hansihe That sounds great! Interested in checking out your fork.

**Comment by miluxmilux** [comment]
Hi @hansihe and @alfg, I also worked on fragmented mp4 this week, and have something working in Safari (that was my main point, since in Firefox and Chrome, things seem to be somewhat less strict.) I haven't worked out the init-segment yet, and my code is definitely not ready for an upstream merge. I only added the `tfdt` box and added `default_sample_(duration|flags|size)` to the `tfhd` box. I would definitely want to check out your code, too, and see what your approach has been. I can maybe show some clean code after the holidays 👍
